### DISCUSSION TITLE
recommended method for extending allowed characters in macro names

### DISCUSSION THREAD

**Discussion by JacobBeaudry** [KaTeX/KaTeX#3325] [Q&A]
Any suggestions as to the best method for extending the allowable characters in custom macro names, with KaTeX? 

[This page](https://texfaq.org/FAQ-linmacnames) on texfaq covers some standard strategies, but IINM, their recommended approach requires `\csname` which is not available with KaTeX. 

I'm looking for a method for defining these macros locally (i.e. on a single page, vs. over the entire site), in case that is relevant. 

Thanks in advance for any suggestions.

**Comment by JacobBeaudry** [comment]
Just wanted to bump this, in hopes of some creative solutions :)

**Comment by edemaine** [comment]
This is controlled by `controlWordRegexString` in `src/lexer.js`. Are you willing to modify the KaTeX source code for your project? If so, this would be the regex to change. There is limited support for catcodes in that file, but not for modifying that regex... I suppose it could be supported (by dynamically changing the lexer regex when these catcodes change) but it wouldn't be easy.

Perhaps you could give an example of a symbol you want to allow in some files but not others.

Note that `\x` already works for any *single* symbol `x`. You just can't currently form multisymbol control words, like LaTeX.
